src/sssom_schema/schema/sssom_schema.yaml

```diff
@@ -657,11 +660,31 @@ classes:
     - object_preprocessing
     - semantic_similarity_score
     - semantic_similarity_measure
+    - similarity_score
+    - similarity_measure
```

Comment:
How do these two slots differ from the `semantic_`-prefixed ones?

Reply:
They are generailisations. Examples of something that is a `similarity_score` but not `semantic` is:

- Levenshtein Distance (Edit Distance): This measures the minimum number of single-character edits (insertions, deletions, or substitutions) required to change one string into another.
- Jaccard Similarity (on strings): This compares the overlap of characters or words between two strings, calculated as the size of the intersection divided by the size of the union of the two sets.
- Hamming Distance: This measures the number of positions at which corresponding characters in two strings of equal length are different.
- Jaro-Winkler Distance: This string metric measures edit distance while giving more favorable ratings to strings that match from the beginning.
- Longest Common Substring: This finds the longest string that is a substring of two or more strings, which can be used as a basis for similarity.

All of these are very typical in ontology / entity matching.

Reply:
Sorry but this seems like another needless complication to me.

The two slots are defined in _exactly the same way_:

```yaml
similarity_score:
  description: A score between 0 and 1 to denote the similarity, where
    1 denotes equivalence.
  range: double
  minimum_value: 0.0
  maximum_value: 1.0
semantic_similarity_score:
  description: A score between 0 and 1 to denote the semantic similarity, where
    1 denotes equivalence.
  range: double
  minimum_value: 0.0
  maximum_value: 1.0
```

The corresponding `similarity_measure` and `semantic_similarity_measure` slots even use the same Jaccard similarity example:

```yaml
semantic_similarity_measure:
  description: The measure used for computing the the semantic similarity score.
    To make processing this field as unambiguous as possible, we recommend using 
    wikidata identifiers, but wikipedia pages could also be acceptable.
  range: string
  examples:
    - value: https://www.wikidata.org/wiki/Q865360
      description: (the Wikidata identifier for the Jaccard index measure).
  similarity_score:
    description: A score between 0 and 1 to denote the similarity, where
      1 denotes equivalence.
    range: double
    minimum_value: 0.0
    maximum_value: 1.0
similarity_measure:
  description: The measure used for computing the the similarity score.
    To make processing this field as unambiguous as possible, we recommend using 
    wikidata identifiers, but wikipedia pages could also be acceptable.
  range: string
  examples:
    - value: https://www.wikidata.org/wiki/Q865360
      description: (the Wikidata identifier for the Jaccard index measure).
```

So, again, why the two different slots? When you defined `semantic_similarity_{measure|score}` you seemingly didn’t have a problem with using it to report a Jaccard similarity, so why do we now need a distinct (non-semantic) `similarity_{measure|score}`?

Reply:
Because there are kinds of jaccard similarity which are semantic similarity measures, like for example "subclass-jaccard" (the number of overlapping parents).

I am happy though for this last push to deprecate the "semantic_*" variants and replace them by the generalised version in a new PR.

Reply:
> I am happy though for this last push to deprecate the "semantic_*" variants and replace them by the generalised version in a new PR.

`semantic_similarity_{score|measure}` have been in the spec for quite some time, so if you do that (I have no objection, to be clear) I would recommend to add a new subsection to the [Compatibility with pre-1.0 versions](https://mapping-commons.github.io/sssom/spec-formats-tsv/#compatibility-with-pre-10-versions) section:

> ## semantic_similarity_score and semantic_similarity_measure
>
> Initial versions of this specification defined a `semantic_similarity_score` slot to store the semantic similarity, and a `semantic_similarity_measure` slot to describe how the the semantic similarity is assessed. In SSSOM 1.0, those slots were replaced by more generic `similarity_score` and `similarity_measure` slots.
>
> Upon encountering a `semantic_similarity_score` (respectively `semantic_similarity_measure`) slot, implementations supporting pre-1.0 versions MUST silently transform it into a `similarity_score` (respectively `similarity_measure`) slot. No changes on the value of the slot are required.

Reply:
#385